workbench-core/base/src/utilities/queryParamsFiltersUtils.ts

```diff
@@ -0,0 +1,110 @@
+/*
+ *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *  SPDX-License-Identifier: Apache-2.0
+ */
+import { marshall } from '@aws-sdk/util-dynamodb';
+import Boom from '@hapi/boom';
+import { QueryParams } from '../aws/helpers/dynamoDB/dynamoDBService';
+import { FilterRequest } from '../interfaces/FilterRequest';
+import { SortRequest } from '../interfaces/SortRequest';
+
+/************************************************************
+ * Validates only one sorting or filter in list requests
+ * @param filter - filter request object
+ * @param sort - sorting request object
+ ************************************************************/
+export function validateSingleSortAndFilter(filter?: FilterRequest, sort?: SortRequest): void {
+  const filterAttributesLength = filter ? Object.keys(filter).length : 0;
+  const sortAttributesLength = sort ? Object.keys(sort).length : 0;
+  // Check that at most one filter is defined because we not support more than one filter
+  if (filterAttributesLength > 1) {
+    throw Boom.badRequest('Cannot apply more than one filter.');
+  }
+  // Check that at most one sort attribute is defined because we not support sorting by more than one attribute
+  if (sortAttributesLength > 1) {
+    throw Boom.badRequest('Cannot sort by more than one attribute.');
+  }
+  const hasSameProperty =
+    filterAttributesLength === 1 &&
+    sortAttributesLength === 1 &&
+    Object.keys(filter ?? {})[0] === Object.keys(sort ?? {})[0];
+  if (filter && sort && !hasSameProperty) {
+    throw Boom.badRequest('Cannot apply a filter and sort at the same time');
+  }
+}
+
+/************************************************************
+ * Parses FilterRequest into QueryParam object consumed by dynamoDB
+ *
+ * @param filter - filter request object
+ * @param gsiPropertyNames - field names on pascal case format to retrieve gsi name
+ *
+ * @returns QueryParams object
+ ************************************************************/
+export function getFilterQueryParams(filter: FilterRequest | undefined, gsiNames: string[]): QueryParams {
+  let queryParams: QueryParams = {};
+  //validate filter request doesnt have more than one filter or no filters
+  if (!filter || Object.keys(filter).length === 0) {
+    return queryParams;
+  }
+  if (Object.keys(filter).length > 1) {
+    throw Boom.badRequest('Cannot filter by more than one attribute.');
+  }
+  //only one entry will exist due to validation above
+  Object.entries(filter).forEach(([key, value]) => {
```

Comment:
Since only one entry is allowed I feel like this is adding unnecessary complication to the function.

Reply:
got it, I tried to allow multiple operators but found out dynamo doesn't allow it and forgot to change it back

Reply:
changed to only retrieve first